R/restore.R

```diff
@@ -25,12 +25,20 @@ isFromCranlikeRepo <- function(pkgRecord, repos) {
     return(TRUE)
   }
 
-  # for records that do declare a source, ensure it's not 'source', 'github', 'bitbucket', or 'gitlab'.
+  # check if there's a remote host for github, bitbucket, or gitlab
+  git_remote <- isBitbucketURL(pkgRecord$remote_host) ||
+    isGitHubURL(pkgRecord$remote_host) ||
+    isGitlabURL(pkgRecord$remote_host)
+  if (git_remote) {
+    return(FALSE)
+  }
```

Comment:
The main downside is that this will do the wrong thing for users with hosted / enterprise instances of these services, when they're accessible from non-default URLs.

Reply:
We could also simply look for the presence of `remote_host` along with maybe `remote_repo`?